src/lua/circllhist.lua

```diff
@@ -0,0 +1,565 @@
+      map['nan'] = count
+    else
+      map[bin] = count
+    end
+  end
+  return map
+end
+
+-- returns total number of samples in histogram
+function Histogram_C:count()
+  return libhist.hist_sample_count(self)
+end
+
+function Histogram_C:stddev()
+  return libhist.hist_approx_stddev(self)
+end
+
+function Histogram_C:moment(k)
+  return libhist.hist_approx_moment(self, k)
+end
+
+--- calculate quantiles over a histogram
+-- quantiles have to be sorted in ascending order
+-- quantiles are-non unique. We return the lowest valid quantile.
+-- returns nil if the histogram is empty
+function Histogram_C:quantiles(...)
+  return unpack(self:_quantiles({...}))
+end
+
+-- returns ratios of samples that are lower than the provided values.
+-- E.g.
+-- a, b = inverse_quantiles(10, 15)
+-- a = ratio of samples <10
+-- b = ratio of samples <15
+function Histogram_C:inverse_quantiles(...)
+  return unpack(self:_inverse_quantiles({...}))
+end
+
+function Histogram_C:_inverse_quantiles(thresholds)
+  for i = 2, #thresholds do
+    if thresholds[i-1] > thresholds[i] then error { "Threshold values need to be in ascending order." } end
+  end
+
+  local total = 0 -- count non nans
+  for _, bucket_bound, count in self:funiter():filter(vc_not_nan) do
+    total = total + count
+  end
+
+  if total == 0 then
+    return {} -- no values for empty histogram
+  end
+
+  local ivq_iter = coroutine.wrap(function(val)
+      local subtotal = 0
+      for _, bucket_bound, count in self:funiter():filter(vc_not_nan) do
+        local bucket_size = Histogram_C.bucket_size(bucket_bound)
+        local bucket_left, bucket_right
+        if bucket_bound < 0 then
+          bucket_left = bucket_bound - bucket_size
+          bucket_right = bucket_bound
+        else
+          bucket_left = bucket_bound
+          bucket_right = bucket_bound + bucket_size
+        end
+        while val < bucket_left do
+          val = coroutine.yield(subtotal/total)
+        end
+        while bucket_left <= val and val <= bucket_right do
+          if bucket_size > 0 then
+            -- value in bucket
+            val = coroutine.yield((subtotal + count * (val - bucket_left) / bucket_size) / total)
+          else  -- the 0-bucket case
+            val = coroutine.yield(subtotal / total)
+          end
+        end
+        if bucket_right < val then
+          -- value larger than bucket
+          subtotal = subtotal + count
+        end
+      end
+      -- value larger than all buckets
+      while val do
+        val = coroutine.yield(1)
+      end
+  end)
+
+  local out = {}
+  for i = 1, #thresholds do
+    out[#out + 1] = ivq_iter(thresholds[i])
+  end
+  return out
+end
+
+--- returns the number of values in a histogram that are in buckets
+--- that are entirely lower or equal than a given value.
+function Histogram_C:count_below(...)
+  local out = {}
+  for i = 1, select("#", ...) do
+    out[i] = self:_count_below_single(select(i, ...))
+  end
+  return unpack(out)
```

Comment:
We create a table just to unpack it below. Is much faster to do this via recursion.

Also why do we want to support a multi argument `count_below()` in the API? We only have a single argument version in C. This is a pure convenience feature.

Reply:
Similar arguments apply to count_above, count_nearby, and others.